contracts/solver-registry/src/pool.rs

```diff
@@ -0,0 +1,199 @@
+            "Not enough attached deposit"
+        );
+
+        // Get new pool ID
+        let pool_id = self.pools.len();
+
+        // Create sub account for managing liquidity pool's assets in NEAR Intents
+        let pool_account_id = self.get_pool_account_id(pool_id);
+        Promise::new(pool_account_id.clone())
+            .create_account()
+            .transfer(CREATE_POOL_STORAGE_DEPOSIT)
+            .deploy_contract(include_bytes!("../../intents-vault/res/intents_vault.wasm").to_vec())
+            .then(
+                Self::ext(env::current_account_id())
+                    .with_static_gas(GAS_CREATE_POOL_CALLBACK)
+                    .on_create_liquidity_pool_account(pool_id, token_ids, fee),
+            )
+            .into()
+    }
+
+    #[private]
+    pub fn on_create_liquidity_pool_account(
+        &mut self,
+        pool_id: u32,
+        token_ids: Vec<AccountId>,
+        fee: u32,
+        #[callback_result] call_result: Result<(), PromiseError>,
+    ) -> Option<u32> {
+        if call_result.is_err() {
+            None
+        } else {
+            // Add the new liquidity pool
+            let pool = Pool::new(token_ids.clone(), fee);
+            self.pools.push(pool);
+            self.pools.flush();
+
+            Event::CreateLiquidityPool {
+                pool_id: &pool_id,
+                token_ids: &token_ids,
+                fee: &fee,
+            }
+            .emit();
+
+            Some(pool_id)
+        }
+    }
+
+    #[private]
+    pub fn on_deposit_into_pool(
+        &mut self,
+        amount: U128,
+        #[callback_result] used_fund: Result<U128, PromiseError>,
+    ) -> U128 {
+        if let Ok(used_fund) = used_fund {
+            // Refund the unused amount.
+            // ft_transfser_call() returns the used fund
+            U128(amount.0.saturating_sub(used_fund.0))
+        } else {
+            amount
+        }
+    }
+
+    // `add_liquidity` and `remove_liquidity` are not needed for now
+    // #[payable]
+    // pub fn add_liquidity(
+    //     &mut self,
+    //     pool_id: u32,
+    //     token_ids: Vec<AccountId>,
+    //     amounts: Vec<Balance>,
+    // ) {
+    //     require!(token_ids.len() == 2, "Must have exactly 2 tokens");
+    //     require!(amounts.len() == 2, "Must have exactly 2 amounts");
+    //     require!(amounts[0] > 0, "Amount must be greater than 0");
+    //     require!(amounts[1] > 0, "Amount must be greater than 0");
+
+    //     let pool = self.pools.get(pool_id).expect("Pool not found");
+    //     let shares_total_supply = pool.shares_total_supply;
+    // }
+
+    // #[payable]
+    // pub fn remove_liquidity(&mut self, pool_id: u32, shares: U128) {
+    //     let shares = shares.0;
+    //     require!(shares > 0, "Shares must be greater than 0");
+
+    //     let pool = self.pools.get(pool_id).expect("Pool not found");
+    //     // pool.shares_total_supply -= shares;
+    // }
+}
```

Comment:
Let's create an issue for allowing users to add liquidity

Reply:
Yes. I have created an issue: https://github.com/Near-One/tee-solver/issues/2